presto-thrift-interface/src/main/java/com/facebook/presto/thrift/interfaces/client/ThriftRangeValueSet.java

```diff
@@ -0,0 +1,183 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.thrift.interfaces.client;
+
+import com.facebook.presto.spi.predicate.Marker;
+import com.facebook.presto.spi.predicate.Range;
+import com.facebook.presto.spi.predicate.SortedRangeSet;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.swift.codec.ThriftConstructor;
+import com.facebook.swift.codec.ThriftEnum;
+import com.facebook.swift.codec.ThriftEnumValue;
+import com.facebook.swift.codec.ThriftField;
+import com.facebook.swift.codec.ThriftStruct;
+
+import javax.annotation.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.facebook.presto.thrift.interfaces.client.ThriftRangeValueSet.ThriftRange.Bound.fromMarkerBound;
+import static com.facebook.presto.thrift.interfaces.client.ThriftRangeValueSet.ThriftRange.toRange;
+import static com.facebook.swift.codec.ThriftField.Requiredness.OPTIONAL;
+import static java.util.stream.Collectors.toList;
+
+@ThriftStruct
+public final class ThriftRangeValueSet
+{
+    private final List<ThriftRange> ranges;
+
+    @ThriftConstructor
+    public ThriftRangeValueSet(@ThriftField(name = "ranges") List<ThriftRange> ranges)
+    {
+        this.ranges = ranges;
+    }
+
+    @ThriftField(1)
+    public List<ThriftRange> getRanges()
+    {
+        return ranges;
+    }
+
+    public static SortedRangeSet toRangeValueSet(ThriftRangeValueSet valueSet, Type type)
+    {
+        List<Range> ranges = new ArrayList<>(valueSet.getRanges().size());
+        for (ThriftRange thriftRange : valueSet.getRanges()) {
+            ranges.add(toRange(thriftRange, type));
+        }
+        return SortedRangeSet.copyOf(type, ranges);
+    }
+
+    public static ThriftRangeValueSet fromSortedRangeSet(SortedRangeSet valueSet)
+    {
+        List<ThriftRange> ranges = valueSet.getOrderedRanges()
+                .stream()
+                .map(ThriftRange::fromRange)
+                .collect(toList());
+        return new ThriftRangeValueSet(ranges);
+    }
+
+    @ThriftStruct
+    public static final class ThriftRange
+    {
+        @ThriftEnum
+        public enum Bound
+        {
+            BELOW(1),   // lower than the value, but infinitesimally close to the value
+            EXACTLY(2), // exactly the value
+            ABOVE(3);   // higher than the value, but infinitesimally close to the value
+
+            private final int value;
+
+            Bound(int value)
+            {
+                this.value = value;
+            }
+
+            @ThriftEnumValue
+            public int getValue()
+            {
+                return value;
+            }
+
+            public static Bound fromMarkerBound(Marker.Bound bound)
+            {
+                switch (bound) {
```

Comment:
Only thrift's `Bound` is imported as `Bound`. The variable that is used for switch doesn't require prefixing enum values with full names. This is how it works.